pkg/registry/registry.go

```diff
@@ -221,6 +221,73 @@ func RegistryLabeller(req *http.Request, resp *http.Response) prometheus.Labels
 	return labels
 }
 
+func (r PlainRegistry) listArchsWithAuth(ctx context.Context, client http.Client, auth AuthenticationToken, registry, image, tag string) ([]Platform, error) {
+	if registry == "docker.io" {
+		registry = "registry-1." + registry
+	}
+	req, err := newGetManifestRequest(ctx, r.Scheme, registry, image, tag, auth)
+	if err != nil {
+		return nil, err
+	}
+	// https://docs.docker.com/registry/spec/manifest-v2-2/
+	req.Header.Add("Accept", "application/vnd.oci.image.index.v1+json")
+	req.Header.Add("Accept", "application/vnd.docker.distribution.manifest.list.v2+json")
+	req.Header.Add("Accept", "application/vnd.oci.image.manifest.v1+json")
+	req.Header.Add("Accept", "application/vnd.docker.distribution.manifest.v2+json")
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Looking at #84 it seems to me that this block and https://github.com/adevinta/noe/pull/83/files#diff-cb61d8f79451b9541de4a8cc0811523a68d15452b2f5971c7618ea5b423cf4ecR260-R269 are so close to each other that they could be a helper, and simplify the error handling and readability of the function overall.